pkg/config/kube/conversion.go

```diff
@@ -54,6 +54,20 @@ func ConvertProtocol(port int32, portName string, proto corev1.Protocol, appProt
 	name := portName
 	if appProto != nil {
 		name = *appProto
+		// Kubernetes has a few AppProtocol specific standard names defined in the Service spec
+		// Handle these only for AppProtocol (name cannot have these values, anyways).
+		switch name {
```

Comment:
This approach assumes `appProtocol` over TCP only.
I think some values like `kubernetes.io/raw` should work for all port protocols. ([context](https://github.com/kubernetes/kubernetes/pull/115433#discussion_r1135945782))

I think it will be better if we change the `protocol.Parse()` func to get the `appProtocol` name and generate a decision based on the combination of protocol and `appProtocol`.

Reply:
Updated with the new PR. Not sure this fully meets your comments. My expectation is all of this is TCP as we already filter out UDP earlier

Reply:
seems like the purpose of [protocol.Parse()](https://github.com/istio/istio/blob/master/pkg/config/protocol/instance.go#L65:6) is to return the right protocol based on the combination. 

Say k8s introduces a new standard in the future that say implementations not to try and guess the protocol and just treat it as raw TCP/UDP/SCTP (and it has already been discussed) it may be better to use the combination of appProtocol and port protocol to decide how to treat the traffic.

Reply:
it is LGTM for the new protocol added in the KEP so you can go with it now and revisit this code in the future if needed.